martin/src/cog/image.rs

```diff
@@ -0,0 +1,234 @@
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::Path;
+
+use martin_tile_utils::TileCoord;
+use tiff::decoder::{Decoder, DecodingResult};
+
+use super::CogError;
+use crate::{MartinResult, TileData};
+
+/// Image represents a single image in a COG file. A tiff file may contain many images.
+/// This type contains several useful information and method for taking tiles from the image.
+#[derive(Clone, Debug)]
+pub struct Image {
+    /// IFD(Image file directory) number.
+    /// An IFD contains information about the image, as well as pointers to the actual image data.
+    pub ifd: usize,
```

Comment:
This is really a bit nit-ty, but I would prefer this being a full variable name.

This way, below code is more clear when read for the first time.

If you disagree, we can just go with IFD, that is fine too.

```suggestion
    /// Contains information about the image, as well as pointers to the actual image data.
    /// Sometimes abbreviated as IFD.
    pub image_file_directory: usize,
```